api/apps/api/src/modules/async-jobs-garbage-collector/garbage-collect-async-jobs.handler.ts

```diff
@@ -0,0 +1,59 @@
+import {
+  CommandHandler,
+  EventBus,
+  IInferredCommandHandler,
+} from '@nestjs/cqrs';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { UsersProjectsApiEntity } from '../access-control/projects-acl/entity/users-projects.api.entity';
+import { UsersScenariosApiEntity } from '../access-control/scenarios-acl/entity/users-scenarios.api.entity';
+import { AsyncJobsGarbageCollectorFinished } from './async-jobs-garbage-collector-finished.event';
+import { GarbageCollectAsyncJobs } from './garbage-collect-async-jobs.command';
+import { ProjectAsyncJobsGarbageCollector } from './project-async-jobs.garbage-collector';
+import { ScenarioAsyncJobsGarbageCollector } from './scenario-async-jobs.garbage-collector';
+
+@CommandHandler(GarbageCollectAsyncJobs)
+export class GarbageCollectAsyncJobsHandler
+  implements IInferredCommandHandler<GarbageCollectAsyncJobs> {
+  constructor(
+    @InjectRepository(UsersProjectsApiEntity)
+    private readonly usersProjectsRepo: Repository<UsersProjectsApiEntity>,
+    @InjectRepository(UsersScenariosApiEntity)
+    private readonly usersScenariosRepo: Repository<UsersScenariosApiEntity>,
+    private readonly projectAsyncJobs: ProjectAsyncJobsGarbageCollector,
+    private readonly scenarioAsyncJobs: ScenarioAsyncJobsGarbageCollector,
+    private readonly eventBus: EventBus,
+  ) {}
+
+  async execute({ userId }: GarbageCollectAsyncJobs) {
+    const projectsIds = await this.getProjectsIdsOf(userId);
+    await Promise.all(
+      projectsIds.map((projectId) =>
+        this.projectAsyncJobs.sendFailedApiEventsForStuckAsyncJobs(projectId),
+      ),
+    );
+
+    const scenariosIds = await this.getScenariosIdsOf(userId);
+    await Promise.all(
+      scenariosIds.map((scenarioId) =>
+        this.scenarioAsyncJobs.sendFailedApiEventsForStuckAsyncJobs(scenarioId),
+      ),
+    );
```

Comment:
this is very tempting, and is ok - thinking however about broader use of the platform later on, I wonder if we could trade some of the performance gains we can hope when count of projects and scenarios is rather low (by firing all the promises at once waiting for all the async work to be done) for some control over potential db query stampedes, and use a `for await...` here instead (accepting to process each project and scenario sequentially)

Reply:
I 

> thanks @angelhigueraacid - great PR 🙌🏼
> 
> my only concern is about trying to do too much at once - see my inline notes on whether we should consider processing each project and scenario check sequentially, and each job kind check within each of them sequentially too, since we shouldn't be too concerned about the checks completing as soon as possible (and conversely if we unleash potentially hundreds of checks at very short interval on a large `api_events` table, we may actually overload the db 🤔

I will gladly change the code to process project and scenarios one by one and each async job kind within each of the sequentially. I had this same concern and i believe this change will make it safer. Thanks for the suggestion